gpm_api/tests/io/test_checks.py

```diff
@@ -0,0 +1,478 @@
+
+
+def test_check_groups() -> None:
+    """ Test check_groups()
+
+    Similar logic to check_variables
+    """
+
+    group_list = ['NS', 'HS', 'MS']
+
+    # Check if None, None is returned
+    res = checks.check_groups(None)
+    assert res is None, "None is not returned"
+
+    # Check if string, string is returned
+    res = checks.check_groups(group_list[0])
+    assert res == [group_list[0]], "String is not returned"
+
+    # Check if list, list is returned
+    res = checks.check_groups(group_list)
+    assert isinstance(res, np.ndarray), "Array is not returned"
+
+    for group in group_list:
+        assert group in res, f"Group '{group}' is not returned"
+
+    # Check if numpy array, list is returned
+    group_list_ndarray = np.array(group_list)
+    res = checks.check_groups(group_list_ndarray)
+    assert isinstance(res, np.ndarray), "numpy array is not returned"
+    assert np.array_equal(res, group_list_ndarray), "Return not equal to input"
+
+    # Check if not list or string, TypeError is raised
+    with pytest.raises(TypeError):
+        checks.check_groups(123)
+
+
+def test_check_version(
+    versions: list[int],
+) -> None:
+    """ Test check_version()
+
+    Possible versions are integers of 4-7
+    """
+
+    # Check if None, None is returned
+    with pytest.raises(ValueError):
+        res = checks.check_version(None)
+
+    # Check if string, exception is raised
+    with pytest.raises(ValueError):
+        checks.check_version('6A')
+
+    # Check if outside range
+    with pytest.raises(ValueError):
+        checks.check_version(123)
+
+    # Check available range should not raise exception
+    for version in versions:
+        res = checks.check_version(version)
+        assert res is None, (
+            f"Function returned {res} for version {version}, expected None"
+        )
+
+    # Try versions outside of range
+    for version in list(range(0, 3)) + list(range(8, 10)):
+        with pytest.raises(ValueError):
+            checks.check_version(version)
+
+
+def test_check_product(
+    product_types: list[str],
+) -> None:
+    """ Test check_product()
+
+    Depends on available_products(), test ambiguous product names similar to
+    those that exist
+    """
+
+    # Test a product that does exist
+    for product_type in product_types:
+        for product in available_products(product_type=product_type):
+            res = checks.check_product(product, product_type=product_type)
+            assert res is None, (
+                f"Function returned {res} for product {product}, "
+                f"expected None"
+            )
+
+    # Test a product that isn't a string
+    for product_type in product_types:
+        for product in [('IMERG'), 123, None]:
+            with pytest.raises(ValueError):
+                checks.check_product(product, product_type=product_type)
+
+
+def test_check_product_type(
+    product_types: list[str],
+) -> None:
+    ''' Test check_product_type()'''
```

Comment:
Is black and ruff running on these tests? I think does not allow trailing space at sentence start ... and should also convert  `'` to `"`

Reply:
No, sorry I wasn't using any. I'll switch on black